### ISSUE TITLE
KIlipperScreen & Enable a read-only filesystem.

### ISSUE THREAD

**Issue by cptcl** [leadustin/QIDI-up2date-german#4]
Hallo und danke erstmal für deinen ausführlichen Guide

Wollte fragen ob du denkst, dass es machbar wäre, auf den RPi Klipperscreen zu installieren und danach das Filesystem auf Read-Only zu stellen? Soweit ich das verstanden habe, braucht es den RPi ja nur um das Display zu steuern sowie Ein und Ausgaben zwischen Display und Drucker zu übermitteln.

Meine Idee wäre das Filesystem auf Read-Only zu stellen und somit das problem des des "Hart Ausschaltens" und schäden am Datei system zu riskieren zu vermeiden. Oder meinst du wird die SD-Karte auf dauer Physich beschädigt durch das hart ausschalten?

Würde mir da eine eher automatischere Lösung wünschen.
Meine Gedanken hierzu...
1. Dachte da an 2x 
``` https://www.berrybase.at/shelly-plus-plug-s-schmale-bluetooth-wlan-steckdose-mit-messfunktion-und-rgb-led-schwarz ```
Jedoch benutze ich keinen Homeassistent wie Alexa oder sowas um die anzusteuern. Und Handyapp zum ausschalten... naja.. kein großer Fan davon.
2. Oder geht der RPi / Display dann in den "Sleep" modus mit display aus wenn der Drucker nicht mehr läuft? (Und,.. kommt da auch wieder raus, wenn der Drucker eingeschalten wird?)
Hätte auch kein problem damit wenn ich nur den Drucker ausschalte wenn dem so wäre, dann nutze ich den RPi noch für andere zwecke ausser KlipperScreen, eventuell webserver für eine kleine HP oder so. Sollte ein RPi 5 eigentlich von der Leistung her locker schaffen.
3. Ist dein Script / Macro dann auf dem Display sichtbar zu machen? Schaltet sich der RPi  dann wieder ein wenn zb der Drucker und RPi beide vom Strom genommen sind (also ich drücke das Makro, warte kurz und dann Knips -> Aus -> Steckdosenleiste) und dann wieder zugeführt werden? (Knips -> An -> Steckdosenleiste)

Denn... zzt schalte ich den Drucker per Knips an der Steckdosenleiste ein und aus.

Deine Meinung und Erfahrung zu einer vernünftigen Lösung hierzu wären sehr erwünscht
PS: RPi, Display, sowie alle Kabel und Netzadapter wurden bestellt und kommen in den nächsten Tagen

**Comment by leadustin** [comment]
Sorry bin bis heute Abend unterwegs. Melde mich.

**Comment by leadustin** [comment]
> Wollte fragen ob du denkst, dass es machbar wäre, auf den RPi Klipperscreen zu installieren und danach das Filesystem auf Read-Only zu stellen? Soweit ich das verstanden habe, braucht es den RPi ja nur um das Display zu steuern sowie Ein und Ausgaben zwischen Display und Drucker zu übermitteln.

Davon würde ich abraten, da es Probleme beim aktualisieren der Software geben wird.

Bzgl. deiner anderen Fragen. Mit dem Shelly Plug habe ich keine Erfahrung. Du kannst aber Klipperscreen so konfigurieren, das es nach einer definierten Zeit das Display ausstellt. Per Tippen des Screens wird es wieder aktiviert. Wenn du den Drucker stromlos machst, wird Klipperscreen für einen bestimmten Zeitraum versuchen Moonraker zu erreichen. Allerdings wird dies nach einer bestimmten Anzahl von Versuchen abgebrochen. Man könnte das sicherlich anpassen, ob es sinnvoll ist steht auf einem anderen Blatt.

Bzgl. meines Macros. Damit wird nur der Raspberry heruntergefahren. Es gibt jedoch einen Weg mit einer Tasmota Steckdose das so zu automatisieren, das der Pi und der Drucker nach dem Druckende heruntergefahren wird.. Dazu müsste man das Script zum herunterfahren des Pis in einem anderen Script ausführen. Allerdings geht es damit nur der Drucker aus. Zum einschalten müsstest du dann die Tasmota-App nutzen. 

Daher bin ich bei eine Sprachsteuerung wie Alexa hängengeblieben. Macro in Mainsail ausführen, Drucker-Hoste in Mainsail herunterfahren und per Sprache dann den Drucker uhnd Pi stromlos machen. Du kannst auch am Display über Klipperscreen den Drucker und Klipperscreen selbst herunterfahren. Musst also nicht zwingend am PC sein.

**Comment by cptcl** [comment]
Vielen Dank. Ja.. Also dann wären meine beiden Optionen wohl.. Den pi einfach laufen zu lassen.. Und was für mich auch OK wäre.
Oder den pi am Display auszuschalten und dann den Drucker.
Und den pi so einstellen, dass er sich selbst startet sobald er dann wieder Strom bekommt. In etwa richtig so oder?
Oder eben mit ner alexa
Will eben nicht den pi dann mit irgendwas extra anschalten müssen. Dann wäre es mir lieber er läuft durch.

**Comment by leadustin** [comment]
> Und den pi so einstellen, dass er sich selbst startet sobald er dann wieder Strom bekommt. In etwa richtig so oder?

Der Pi startet eh sobald er Strom bekommt

Am elegantesten ist es m.M.n. per Sprachsteuerung. Wenn du am Drucker bist, kannst du den Pi und das System des Druckers über Klipperscreen herunterfahren. Bist du am PC oder Handy dann über Mainsail/Fluidd per Macro und Host. Zum Schluss halt "Alexa Drucker aus".

**Comment by cptcl** [comment]
Vielen Dank für die Infos. Display und pi sind heute gekommen. 
Das alte display bekommt man jedoch nicht weg ohne das ganze ding aufzuschrauben oder? Hab gestern nachgesehen,.. glaube nicht dass man da ohne viel zerlegen an die oberen schrauben kommt.

**Comment by leadustin** [comment]
> Kann ich den Pi auch über usb mit dem board connecten statt wlan / lan?

Ist mir nicht bekannt ob das funktioniert. Denke out of the box nicht, da Klipperscreen ja über die KlipperScreen.conf die Verbindung zu Moonraker herstellt. Mit diesem Prinzip kannst du auch mehrere Drucker über das eine Display steuern.

**Comment by cptcl** [comment]
Stimmt. Danke für deinen input. Denke auf kurz oder lang wirdes zu dem total umbau bei mir kommen. Qidi hat zwar gesagt sie arbeiten an einer 0.12 klipper version für den X-Max3 aber wenn die dann wieder 3 jahre brauchen für zb Klipper 0.15 oder so... naja. Verscuhe noch die entwickler von Klipperscreen dazu zu bringen, den Filament typ in Klipperscreen mitanzeigen zu lassen. Ist so das einzige was mir hier echt fehlt. (ZB wenn man den Filament typ nicht im Namen mit rein schreibt, weisst du nach einiger zeit ja nicht mehr was das war.
Nun... ich bin jetzt nur noch auf der suche wie ich DPMS hinbekomme, damit der bildschirm nicht durchgehend läuft.
